lib/cacheable/response_cache_handler.rb

```diff
@@ -151,8 +154,7 @@ def page_too_old(timestamp, cache_age_tolerance)
 
     def cache_return!(message, &block)
       Cacheable.log message
-      @controller.instance_eval(&block)
-      throw :cache_return
+      throw :cache_return, block.call
```

Comment:
There's no reason for this to take a block anymore, I'm going to change it into taking a response and passing it back.